Src/Test.UnitTests.Sarif.PatternMatcher.Cli/AnalyzeCommandTests.cs

```diff
@@ -453,6 +548,124 @@ private SarifLog RunAnalyzeCommand(string definitionsText, string fileContents)
+
+            mockFileSystem.Setup(x => x.FileInfoLength(SmallTargetName)).Returns(fileContents.Length);
+            mockFileSystem.Setup(x => x.FileExists(@$"c:\Test.UnitTests.Sarif.PatternMatcher.Cli.dll")).Returns(true);
+            mockFileSystem.Setup(x => x.AssemblyLoadFrom(It.IsAny<string>())).Returns(Assembly.LoadFrom(dllLocation));
+
+            Program.FileSystem = mockFileSystem.Object;
+
+            string tempFileName = Path.GetTempFileName();
+            string sarifLogFileName = $"{tempFileName}.sarif";
+            SarifLog sarifLog = null;
+            string levels = "Error;Warning;Note";
+
+            try
+            {
+                string[] args = new[]
+                {
+                    "analyze",
+                    scanTargetPath,
+                    $"-d", searchDefinitionsPath,
+                    $"-o", sarifLogFileName,
+                    "--level", levels,
+                    "--rich-return-code",
+                };
+
+                if (runDynamicValidation)
+                {
+                    args.Append("--dynamic-validation");
+                }
+
+                int result = Program.Main(args);
+                result.Should().Be(0);
+
+                sarifLog = JsonConvert.DeserializeObject<SarifLog>(File.ReadAllText(sarifLogFileName));
+            }
+            finally
+            {
+                if (File.Exists(tempFileName))
+                {
+                    File.Delete(tempFileName);
+                }
+
+                if (File.Exists(sarifLogFileName))
+                {
+                    File.Delete(sarifLogFileName);
+                }
+            }
+
+            return sarifLog;
+        }
+
+        private void ValidateFailureLevelByValidationType(
+            FailureLevel expectedFailureLevel,
+            SarifLog sarifLog,
+            string validationScenario,
+            bool isDynamicAnalysis,
+            ref StringBuilder stringBuilder)
+        {
+            string testScenarioMode = isDynamicAnalysis ?
+                    "with dynamic validation enabled" :
+                    "without dynamic validation enabled";
+
+            if (sarifLog == null)
+            {
+                string message = $"SARIF result should not be null for `{validationScenario}` test scenario {testScenarioMode}.";
+                StringBuilderFormatAndAppendNewLine(message, ref stringBuilder);
+            }
+            else if (sarifLog.Runs[0]?.Results[0]?.Level != expectedFailureLevel)
+            {
+                string message = $"Expected `FailureLevel` to be `{expectedFailureLevel}` but found " +
+                    $"{sarifLog.Runs[0]?.Results[0]?.Level} for `{validationScenario}` test scenario {testScenarioMode}.";
+
+                StringBuilderFormatAndAppendNewLine(message, ref stringBuilder);
+            }
+        }
+
+        private void StringBuilderFormatAndAppendNewLine(string data, ref StringBuilder stringBuilder)
```

Comment:
>ref [](http://example.com/codeflow?start=70&length=3)

why is this a ref arg? it doesn't need to be. you don't test it for null (and create one if it is null).